Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1212,9 +1212,9 @@ ExceptionOr<Ref<ElementInternals>> HTMLElement::attachInternals()
     return ElementInternals::create(*this);
 }
 
-static ExceptionOr<void> checkPopoverValidity(Element& element, PopoverVisibilityState expectedState)
+static ExceptionOr<void> checkPopoverValidity(HTMLElement& element, PopoverVisibilityState expectedState)
 {
-    if (!element.hasAttributeWithoutSynchronization(HTMLNames::popoverAttr))
+    if (!element.popoverData() || element.popoverState() == PopoverState::None)
```

Comment:
The popoverState() check should be enough, see the definition: https://searchfox.org/wubkat/source/Source/WebCore/html/HTMLElement.cpp#1482